### PR TITLE
various changes

### LINKED ISSUE #39: Issues with displaying annotations out of visible range
Hi !
I just installed this package from melpa, and it gets emacs into a strange state.

when I add an annotation, and the current buffer does not show enough columns (e.g. my view has 62 colums visible), then adding an annotation fails with this message (from *Message* buffer):
Error during redisplay: (jit-lock-function 3042) signaled (args-out-of-range "where it all started" 0 -22)

Note how 85-62 = 23, and we get -22 (I guess it's zero-based).

And the annotation does not show. If I resize then buffer to make space, it displays it, so it gets stored.

When annotations are added when there is enough space, and then the buffer is resized, the annotation flows fine. Just at creation time it gets confused.

### PR DESCRIPTION
**Pull request by cage2**
trying to fix #39 and #33, plus some more changes.

Bye!
C.

### CODE REVIEW

File: README.md
Comment (bastibe) [inline README.md]: This doesn't work, I'm sorry. If you submit a pull request, you can't have copyright of it.

You can add your changes to the changelog under your name, but there can only be one copyright holder.

File: README.md
Comment (bastibe) [inline README.md]: Please do not add extraneous white space.

File: README.md
Comment (bastibe) [inline README.md]: Please do not delete these white spaces. They are meaningful in markdown (also applies to all following lines in this file).

File: annotate.el
Comment (bastibe) [inline annotate.el]: This docstring seems wrong?

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el]: All public names should be prefixed with `annotate-`.

File: annotate.el
Comment (bastibe) [inline annotate.el:10]: As this is more than a bugfix, it should bump to 0.5.0 (first number for breaking changes, second number for new features, third number for bug fixes).

**Review by bastibe — CHANGES_REQUESTED** [review]
Thank you very much for your pull request! Fixing these bugs is very welcome indeed!

However, I've added a few remarks that need to be addressed before merging.

I have to admit, though, it has been a long time since I last worked on this project, and I couldn't follow every code change in detail. I trust that you have tested the code and that it doesn't break functionality.

In general, version number changes and changelog changes are typically left to the maintainer, and should not be part of a pull request.

File: README.md
Comment (cage2) [reply README.md]: Oh, i am sorry to hear that :(

I can not remove the copyright, but i am going to fix the fork little by little so you can merge the PR if you change your mind! :)

Anyway thank you for your review, it was very useful to learn how to write an emacs package.

Bye!
C.

Comment (bastibe) [reply README.md]: This source code is published under the terms of the MIT license, which grants every user to the right to

> use, copy, modify, merge, publish, distribute, sublicense, and/or sell
> copies of the Software, and to permit persons to whom the Software is
> furnished to do so, subject to the following conditions:
> 
> The above copyright notice and this permission notice shall be included in
> all copies or substantial portions of the Software.

(See the [LICENSE](https://github.com/bastibe/annotate.el/blob/master/LICENSE) file)

As you can see, you can do as you please with this software, so long as you accept my authorship of it. Thus, the one thing you can not do is to claim copyright of it, which would give you the right to alter its license.

Your contribution will be duly noted in the changelog, and be visible in the Git history. I can even give you commit rights, if you would like to become the maintainer of this software. But even that would not give you the right to claim copyright of it, as you are not the original author.

I will only be able to merge this pull request if you remove that claim. Which is nothing personal, you understand. I approve of your work, and you seem to be a fine human being. But you are not the author of this software, and therefore can not claim copyright for it.

**Comment by cage2** [comment]
On Wed, Sep 11, 2019 at 05:16:45AM -0700, Bastian Bechtold wrote:
> bastibe commented on this pull request.

Hi!

First, i think is  important to clarify that i am  not upset, angry or
something  like that,  i totally  understand that  it's up  to you  to
accept  or refuse  any  PR for  this  project.  I  would  be happy  to
contribute  to  this package  for  the  public  good,  but i  am  also
satisfied that i  was able to learn  a bit of elisp working  on a code
small and clean enough i could wrap my head around.

>
> > @@ -28,65 +28,72 @@ Alternatively, they can be integrated `annotate-integrate-annotations` as commen
>
>  - annotations in org-mode source blocks will be underlined, but the annotations don't show up. This is likely a fundamental incompatibility with the way source blocks are highlighted and the way annotations are displayed.
>
> +Portion of the code © 2019 Universita' degli Studi di Palermo
> +

>  This source code  is published under the terms of  the MIT license,

[...]

> As you can see, you can do as you please with this software, so long
> as you accept my  authorship of it. Thus, the one  thing you can not
> do is to  claim copyright of it,  which would give you  the right to
> alter its license.

I think there is a misunderstanding here, i do not claim copyright for
your code, this  would be unfair, i claim copyright  just for the code
that are part of this PR.  I  do not expect neither to substitute your
copyright nor to  assert that i am  the original author in  any way. I
just  ask  to add  my  copyright  note  to  the already  existing  one
(i.e. your). Probably i have worded that wrong, my fault.

Moreover i  feel a moral (and  likely legal too) obligation  to assign
the copyright  (again just for the  code i wrote) to  the university i
mentioned,  so  there  is  no  change i  could  remove  the  copyright
attribution, but  of course  it can  be reworded in  a way  that makes
clear that you are original author, that i made just some modification
and  the copyright  is  just for  that specific  portion  of the  code
only. But please i be aware that i am not a lawyer! :)

So said i ask you to reconsider  your decision

Finally i  want to thank  you anyway to  spent your time  reviewing my
code and discuss all that stuff with  me and i hope that there will be
more chance to collaborate in the future.

Bye!
C.

PS:

In an efforts of  transparency i want to point out that  i keep a copy
of       this        project       (with        modification)       on
https://notabug.org/cage/annotate.el if you are not OK with that just
tell me and i will delete or rename the repository.

**Comment by bastibe** [comment]
[Reading up on this](https://opensource.stackexchange.com/questions/5508/what-does-and-contributors-in-the-copyright-byline-imply), it seems that we can indeed have multiple copyrights, without violating the license.

My point is that I do not want to grant the right to change the license.

As such, I think the right way to do this is to add your copyright to the LICENSE file. However, it also seems that no explicit copyright notice is needed and your contributions are always copyrighted to you, as I do not require a copyright assignment.

**Comment by cage2** [comment]
On Wed, Sep 11, 2019 at 11:46:30PM -0700, Bastian Bechtold wrote:

Hi!

> [Reading                            up                            on
> this](https://opensource.stackexchange.com/questions/5508/what-does-and-contributors-in-the-copyright-byline-imply),
> it  seems  that we  can  indeed  have multiple  copyrights,  without
> violating the license.

So, is it OK to you if i remove my current copyright notice in the readme and add a line on the LICENSE file and top comment of the source, like:

``` text
;; Copyright (C) 2015 Bastian Bechtold

;; Copyright (C) 2019 Universita' degli studi di palermo
```

Are you positive about that?

Bye!

PS: Also i am afraid i have introduced  a bug in the code, sorry. The good
news is that i have ready the fix! :)

**Comment by bastibe** [comment]
I think that is a good compromise. It acknowledges all authors, yet make it clear that all authors agree to the terms of the same license.

To express this more formally, how about:

```
Copyright (C) 2015 Bastian Bechtold and contributors:
Naoya Yamashita (2018)
Universita' degli studi di palermo (2019)
```

As far as I can tell, you'd usually just say "and contributors" in the LICENSE file, and name individual contributions in a CHANGELOG file. But either way is fine by me.

**Comment by cage2** [comment]
On Mon, Sep 16, 2019 at 12:41:15AM -0700, Bastian Bechtold wrote:

Hello!

> I think that is a good compromise.

very well! :)


> It acknowledges all authors, yet make it clear that all authors agree to the terms of the same license.
>
> To express this more formally, how about:
>
> ```
> Copyright (C) 2015 Bastian Bechtold and contributors:
> Naoya Yamashita (2018)
> Universita' degli studi di palermo (2019)
> ```
>
> As far as I can tell, you'd usually just say "and contributors" in the LICENSE file, and name individual contributions in a CHANGELOG file. But either way is fine by me.

I commited  the latest modification as above;  i just capitalized
the name of the university to follow italian grammar rules.

Bye!
C.

PS : I have some more code but i think this is matter for a new PR! :)

**Comment by bastibe** [comment]
Great! I'm glad we found a good solution here.

As far as I can tell, there is still one copyright notice in the README that should be removed. Otherwise, this looks ready to merge to me. Please let me know once you think it's ready to merge.

**Comment by cage2** [comment]
On Mon, Sep 16, 2019 at 11:52:18PM -0700, Bastian Bechtold wrote:
> Great! I'm glad we found a good solution here.

me too!

> As far as I can tell, there is still one copyright notice in the README that should be removed. Otherwise, this looks ready to merge to me. Please let me know once you think it's ready to merge.

ohhh, sorry! I forgot to remove that  line. Fixed! :) I hope i did not
screw up the formatting this time! :)

Bye!
C.

**Comment by bastibe** [comment]
Thank you very much for your contribution!

**Comment by cage2** [comment]
On Wed, Sep 18, 2019 at 12:15:35AM -0700, Bastian Bechtold wrote:
> Thank you very much for your contribution!

my pleasure! :)

Bye!
C.

**Pull request merged**
